src/main/java/org/takes/rq/RqMultipart.java

```diff
@@ -214,28 +191,65 @@ private Request make(final byte[] body, final int start,
     }
 
     /**
-     * Find position of array inside another array.
-     * @param outer Big array
-     * @param inner Small array
-     * @param start Where to start searching
-     * @return Position
+     * Write header.
+     * @param fos Output stream
      * @throws IOException If fails
      */
-    private static int indexOf(final byte[] outer, final byte[] inner,
-        final int start) throws IOException {
-        for (int idx = start; idx < outer.length - inner.length; ++idx) {
-            boolean found = true;
-            for (int sub = 0; sub < inner.length; ++sub) {
-                if (outer[idx + sub] != inner[sub]) {
-                    found = false;
+    private void writeHeader(final OutputStream fos) throws IOException {
+        fos.write(this.head().iterator().next().getBytes("UTF-8"));
+        fos.write("\r\n".getBytes());
+    }
+
+    /**
+     * Multi-part request parsing.
+     * @param matcher Boundary matcher
+     * @param body Request
+     * @return Request collection
+     * @throws IOException If fails
+     * @checkstyle ExecutableStatementCountCheck (40 lines)
+     * @todo #115 Need to refactoring this function to avoid
+     *  ExecutableStatementCountCheck.
+     */
+    private Collection<Request> requests(final Matcher matcher,
+            final InputStream body) throws IOException {
+        final Collection<Request> requests = new LinkedList<Request>();
+        final byte[] boundary = new StringBuilder()
+            .append("\r\n--")
+            .append(matcher.group(1)).toString().getBytes();
+        int pos = 2;
+        File file = null;
+        OutputStream fos = null;
+        while (body.available() > 0) {
```

Comment:
@d-molotchko this loop has terrible design, we should extract this to other object